### PR TITLE
Initial TDX implementation

### PR DESCRIPTION
**Pull request by rbradford**
With a stock TDVF this boots sufficiently far to show that the limitation is in the TDVF code (expecting a Q35 platform) rather than the VMM. (With TDX once the vCPUs are running there is no more interaction from the VMM.

See: #2226

### CODE REVIEW

File: vmm/src/vm.rs [hunk 1]
```diff
@@ -464,7 +464,7 @@ pub fn physical_bits(max_phys_bits: Option<u8>) -> u8 {
 }
 
 pub struct Vm {
-    kernel: File,
+    kernel: Option<File>,
```
Comment (sameo) [inline vmm/src/vm.rs:491]: Should we squash 3476759f0 with that one?

File: vmm/src/vm.rs [hunk 2]
```diff
@@ -1504,6 +1538,126 @@ impl Vm {
         }
     }
 
+    #[cfg(feature = "tdx")]
+    fn configure_tdx(&mut self) -> Result<()> {
```
Comment (sameo) [inline vmm/src/vm.rs:1542]: Nit: I would rename that one to `tdx_init`, in order to more clearly match the underlying ioctl.

**Review by sameo — APPROVED** [review]
Mostly nits, the PR looks really good as usual. Thanks @rbradford

**Review by rbradford — COMMENTED** [review]
Thanks @sameo I think I integrated all your feedback along with spotting a few other things.

**Comment by sboeuf** [comment]
I'm a little late to the party, but reading this PR was very informative :+1: 
And thanks to the way it was split, it was quite easy to understand the different bits :)

**Pull request merged**